tsl/test/expected/dist_ddl.out

```diff
@@ -3384,6 +3384,105 @@ SELECT * from show_chunks('dist_test');
+ create_distributed_hypertable 
+-------------------------------
+ (37,public,conditions_dist,t)
+(1 row)
+
+INSERT INTO conditions_dist(day, temperature)
+SELECT ts, date_part('month', ts)*100 + date_part('day', ts)
+FROM generate_series('2010-01-01 00:00:00 MSK' :: timestamptz, '2010-03-01 00:00:00 MSK' :: timestamptz - interval '1 day', '1 day') as ts;
+CREATE MATERIALIZED VIEW conditions_dist_1m
+WITH (timescaledb.continuous, timescaledb.materialized_only = false) AS
+SELECT
+   timescaledb_experimental.time_bucket_ng('1 month', day, 'MSK') AS bucket,
+   MIN(temperature),
+   MAX(temperature)
+FROM conditions_dist
+GROUP BY bucket;
+NOTICE:  refreshing continuous aggregate "conditions_dist_1m"
+SELECT * FROM conditions_dist_1m ORDER BY bucket;
+            bucket            | min | max  
+------------------------------+-----+------
+ Thu Dec 31 13:00:00 2009 PST | 101 | 1231
+ Sun Jan 31 13:00:00 2010 PST | 131 |  227
+(2 rows)
+
+-- case 1: ensure select works when data node is unavailable
+SELECT alter_data_node(:'DATA_NODE_1');
+                 alter_data_node                 
+-------------------------------------------------
+ (db_dist_ddl_1,localhost,55432,db_dist_ddl_1,t)
+(1 row)
+
+SELECT alter_data_node(:'DATA_NODE_1', port => 55433, available => false);
+                 alter_data_node                 
+-------------------------------------------------
+ (db_dist_ddl_1,localhost,55433,db_dist_ddl_1,f)
+(1 row)
+
+SELECT * FROM conditions_dist_1m ORDER BY bucket;
+            bucket            | min | max  
+------------------------------+-----+------
+ Thu Dec 31 13:00:00 2009 PST | 101 | 1231
+ Sun Jan 31 13:00:00 2010 PST | 131 |  227
+(2 rows)
+
+-- case 2: ensure cagg can be updated while data node is not available
+INSERT INTO conditions_dist(day, temperature)
+SELECT ts, date_part('month', ts)*100 + date_part('day', ts)
+FROM generate_series('2010-03-01 00:00:00 MSK' :: timestamptz, '2010-04-01 00:00:00 MSK' :: timestamptz - interval '1 day', '1 day') as ts;
+-- case 3: ensure refresh cagg does not work on a data node
+\set ON_ERROR_STOP 0
+CALL refresh_continuous_aggregate('conditions_dist_1m', '2010-01-01', '2010-04-01');
+ERROR:  could not connect to "db_dist_ddl_1"
+\set ON_ERROR_STOP 1
+-- case 4: ensure that cagg is still updated and data is visible using real-time aggregation
+SELECT * FROM conditions_dist_1m ORDER BY bucket;
+            bucket            | min | max  
+------------------------------+-----+------
+ Thu Dec 31 13:00:00 2009 PST | 101 | 1231
+ Sun Jan 31 13:00:00 2010 PST | 131 |  227
+ Sun Feb 28 13:00:00 2010 PST | 228 |  330
+(3 rows)
+
+-- case 5: TRUNCATE will not work while data node is not available
+\set ON_ERROR_STOP 0
+TRUNCATE TABLE conditions_dist;
+ERROR:  could not connect to "db_dist_ddl_1"
+\set ON_ERROR_STOP 1
+-- case 6: TRUNCATE cagg directly
+\set ON_ERROR_STOP 0
+TRUNCATE conditions_dist_1m;
```

Comment:
Fails because it is trying to send invalidation to DN. Note: data in cagg is stored on AN, so the TRUNCATION itself doesn't fail, but it fails as a result of the failed invalidation.